e2e/js/js-cdn/src/app/app.element.ts

```diff
@@ -0,0 +1,450 @@
+                      fill="none"
+                      stroke="currentColor"
+                      viewBox="0 0 24 24"
+                      xmlns="http://www.w3.org/2000/svg"
+                    >
+                      <path
+                        stroke-linecap="round"
+                        stroke-linejoin="round"
+                        stroke-width="2"
+                        d="M8 9l3 3-3 3m5 0h3M5 20h14a2 2 0 002-2V6a2 2 0 00-2-2H5a2 2 0 00-2 2v12a2 2 0 002 2z"
+                      />
+                    </svg>
+                    View interactive project graph
+                  </summary>
+                  <pre>nx graph</pre>
+                </details>
+                <details>
+                  <summary>
+                    <svg
+                      fill="none"
+                      stroke="currentColor"
+                      viewBox="0 0 24 24"
+                      xmlns="http://www.w3.org/2000/svg"
+                    >
+                      <path
+                        stroke-linecap="round"
+                        stroke-linejoin="round"
+                        stroke-width="2"
+                        d="M8 9l3 3-3 3m5 0h3M5 20h14a2 2 0 002-2V6a2 2 0 00-2-2H5a2 2 0 00-2 2v12a2 2 0 002 2z"
+                      />
+                    </svg>
+                    Run affected commands
+                  </summary>
+                  <pre><span># see what&apos;s been affected by changes</span>
+                    nx affected:graph
+
+                    <span># run tests for current changes</span>
+                    nx affected:test
+
+                    <span># run e2e tests for current changes</span>
+                    nx affected:e2e</pre>
+                </details>
+              </div>
+
+              <p id="love">
+                Carefully crafted with
+                <svg
+                  fill="currentColor"
+                  stroke="none"
+                  viewBox="0 0 24 24"
+                  xmlns="http://www.w3.org/2000/svg"
+                >
+                  <path
+                    stroke-linecap="round"
+                    stroke-linejoin="round"
+                    stroke-width="2"
+                    d="M4.318 6.318a4.5 4.5 0 000 6.364L12 20.364l7.682-7.682a4.5 4.5 0 00-6.364-6.364L12 7.636l-1.318-1.318a4.5 4.5 0 00-6.364 0z"
+                  />
+                </svg>
+              </p>
+            </div>
+          </div>
+    `
+    }
+
+    connectedCallback(): void {
+        this.updateInnerHTML()
+        devcycleClient.subscribe('configUpdated', () => {
+            this.updateInnerHTML()
+        })
+    }
+}
+
+const user = {
+    user_id: 'userId1',
+    email: 'auto@taplytics.com',
+    customData: {
+        cps: 'Matthew',
+        cpn: 777,
+        cpb: true,
+    },
+    isAnonymous: false,
+}
+
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+const devcycleClient = DevCycle.initializeDevCycle(SDK_KEY, user, {
```

Comment:
why's this a TS error?

Reply:
Because the DevCycle variable is coming from the script import so typescript doesn't know it exists. That's why I created the global.d.ts file, but that didn't fully solve the problem either